.pre-commit-config.yaml

```diff
@@ -1,21 +1,23 @@
-# See https://pre-commit.com for more information
-# See https://pre-commit.com/hooks.html for more hooks
+---
+default_language_version:
+  python: python
+
 repos:
--   repo: https://github.com/pre-commit/pre-commit-hooks
+  - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v3.2.0
     hooks:
-    -   id: trailing-whitespace
-    # -   id: end-of-file-fixer
-    -   id: check-yaml
-    -   id: check-added-large-files
--   repo: https://github.com/pycqa/isort
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-added-large-files
+  - repo: https://github.com/pycqa/isort
     rev: 5.12.0
     hooks:
-    -   id: isort
+      - id: isort
```

Comment:
@alexrudd2 all these “- “ to “ -“ changes are you ok with them ? seems like not needed to me.

Reply:
to me it looks as a change just to make a change, but it your turf.

Reply:
our precommit runs a yaml test, and do not complain. If you run a linter locally place make sure it runs with the same options.

Reply:
> @alexrudd2 all these “- “ to “ -“ changes are you ok with them ? seems like not needed to me.

They're fine with me.  The YAML parser is less strict than a YAML validator.

Reply:
> our precommit runs a yaml test, and do not complain. If you run a linter locally place make sure it runs with the same options.

precommit only runs on changed files, I think, so it doesn't run on itself o_0

Reply:
Nor the second or third time...I just made 2 commit, each with a change to  .pre-commit-config.yaml
´´´
(pymodbus) pymodbus % git add .pre-commit-config.yaml 
(pymodbus) pymodbus % git commit
Trim Trailing Whitespace.................................................Passed
Check Yaml...............................................................Passed
Check for added large files..............................................Passed
isort................................................(no files to check)Skipped
black................................................(no files to check)Skipped
[examples 4dbc483] xxx
 1 file changed, 1 deletion(-)
´´´

So now I am all for not changing the file.

Reply:
And just to be sure, I copied the yaml and precommit still do not complain:
```
(pymodbus) pymodbus % cp .pre-commit-config.yaml jan.yaml
(pymodbus) pymodbus % git add jan.yaml
(pymodbus) pymodbus % git commit
Trim Trailing Whitespace.................................................Passed
Check Yaml...............................................................Passed
Check for added large files..............................................Passed
isort................................................(no files to check)Skipped
black................................................(no files to check)Skipped
[examples df50d86] xxx
 1 file changed, 21 insertions(+)
 create mode 100644 jan.yaml
``

There are a ton of linters out there, we do not use, please lets only change files if the linters we use complain.

Reply:
I tried to explain to you do not mix issues in the same pull request, that is the important point. Whether or not they are valuable is something to be discussed outside this PR.

feel free to submit another Pull request with the end_of_file_fixer, then we can discuss that change, which can include a reformatting of pre_commit.

Reply:
Yep I hear that, I made https://github.com/pymodbus-dev/pymodbus/pull/1442

One other comment is in CI, something like `pre-commit run --from-ref origin/dev --to-ref HEAD` can be interesting.

It means `pre-commit` is scoped to that particular PR, not run on files outside of it.  This technique is sometimes called "incremental".  One pro of this is one can make `pre-commit` config tweaks, without having to make repo wide changes (currently mandated by `--all-files`).  One con of this is it can be seen as adding tech debt, since the remainder of the repo is sort of "stale" with respect to the `pre-commit` config.  I am not contributing this, more just an alternate balance I have seen struck.

Reply:
No in CI we try to run on the whole source tree, because we have all to often seen side effects in files not touched.

Locally it needs to be fast so it runs only on the changed files and not with all CI checks.